### DISCUSSION TITLE
include resources in the source code

### DISCUSSION THREAD

**Discussion by febelery** [filamentphp/filament#1298] [Help]
filament use link like:
- `https://ui-avatars.com/api/?name=xxx&color=FFFFFF&background=111827`
-  `https://fonts.gstatic.com/s/dmsans/v6/rP2Cp2ywxg089UriASitCBimCw.woff2`
- `https://fonts.googleapis.com/css2?family=DM+Sans:ital,wght@0,400;0,500;0,700;1,400;1,500;1,700&display=swap`
...

There are some links that don't open in some countries, so the page will keep loading.

So, can include these resources in the source code

**Comment by ManojKiranA** [comment]
You can override the Avatar image by following [setting-up-avatars](https://filamentadmin.com/docs/2.x/admin/users#setting-up-avatars)

here is the **gravatar** implementation

```
public function getFilamentAvatarUrl(): ?string
    {
        return
            Str::of($this->email)
            ->lower()
            ->trim()
            ->pipe('md5')
            ->prepend('https://www.gravatar.com/avatar/');
    }
```
